more_itertools/more.py

```diff
@@ -1321,3 +1324,44 @@ def locate(iterable, pred=bool):
 
     """
     return compress(count(), map(pred, iterable))
+
+
+def lstrip(iterable, skip_items):
+    """Yield the items from *iterable*, but strip any that match *skip_items*
+    from the beginning.
+
+        >>> iterable = (None, False, None, 1, 2, None, 3, False, None)
+        >>> list(lstrip(iterable, (None, False, '')))
+        [1, 2, None, 3, False, None]
+
+    This function is analagous to :func:`str.lstrip`.
+    Note that the *skip_items* argument is not a prefix; any items that it
+    contains will be stripped.
+
+    """
+    return dropwhile(partial(contains, set(skip_items)), iterable)
```

Comment:
You could also use `dropwhile(set(skip_items).__contains__, iterable)`. That could be a bit faster but not much, most of the time will be spent inside the `set.__contains__` function.